wled00/pin_manager.h

```diff
@@ -56,15 +59,29 @@ enum struct PinOwner : uint8_t {
 };
 static_assert(0u == static_cast<uint8_t>(PinOwner::None), "PinOwner::None must be zero, so default array initialization works as expected");
 
+// wled00/cfg.cpp:734:22:   required from here
+// wled00/src/dependencies/json/ArduinoJson-v6.h:4115:25:
+// error: no matching function for call to 'convertToJson(const PinOwner&, ArduinoJson6181_90::VariantRef&)'
+//      return convertToJson(src, dst); // Error here? See https://arduinojson.org/v6/unsupported-set/
+//
+// PROBLEM: Header problems if I try to include headers to have the ArduinoJson::VariantRef type defined here.
+//   CHEAT: Use a template to avoid this, as it won't be instantiated until later, where it's used and the
+//          ArduinoJson::VariantRef type is already defined.
+template <typename T>
```

Comment:
Perhaps I don't understand.  I think this covers all conversions to JSON, and avoids the serialization code needing to `static_cast<>` from `PinOwner` to some other type.  Defining this in this header is good for encapsulation of the "how" of JSON serialization to be right next to the relevant implementation.

Whether that ends up in an array, or as a key's value, is still up to the caller.  The existing code uses it to store it in an array, for example.  Is this not the "right way" to support JSON serialization?

For example, if the decision changes from storing the integer value, to storing a string, (_I think?_) this is the only place that would need to change, and the serialization would suddenly contain strings instead of the integers...